src/hotspot/share/gc/shenandoah/heuristics/shenandoahAdaptiveHeuristics.cpp

```diff
@@ -132,11 +252,15 @@ bool ShenandoahAdaptiveHeuristics::should_start_gc() const {
       return true;
     }
   }
+  return false;
+}
 
+bool ShenandoahAdaptiveHeuristics::is_allocation_rate_too_high(size_t capacity,
+                                                               size_t available,
+                                                               size_t bytes_allocated_since_gc_start) {
   // Check if allocation headroom is still okay. This also factors in:
   //   1. Some space to absorb allocation spikes
   //   2. Accumulated penalties from Degenerated and Full GC
-
   size_t allocation_headroom = available;
 
   size_t spike_headroom = capacity / 100 * ShenandoahAllocSpikeFactor;
```

Comment:
I suspect `ShenandoahAllocSpikeFactor` handling is not needed anymore, as this new code is supposed to provide the similar cushion of few sigmas off the average value. That was my poor-mans solution to the same problem this patch resolves. You can remove the flag and update the comments in this block to mention only accumulated penalties.

Reply:
I'll make some test runs without this flag.